<launch>
    <!--  ************** Global Parameters ***************  -->
    <arg name="trajectory_type" default="straight"
        doc="opt: straight, circle, curve, highstraight, highcurve" />
    <arg name="controller" default="track" doc="opt: open, track, mpc, lqr, pid" />

    <!--  ************** Reference trajectory generation **************  -->
    <node name="trajectory_generate" pkg="scout_unitree_control" type="trajectory_generate">
        <param name="trajectory_type" value="$(arg trajectory_type)" />
    </node>

    <!--  ************** Tracking Node **************  -->
    <node name="scout_opening" pkg="scout_unitree_control" type="scout_opening" output="screen"
        if="$(eval controller == 'open')">
    </node>
    <node name="scout_tracking" pkg="scout_unitree_control" type="scout_tracking" output="screen"
        if="$(eval controller == 'track')">
        <rosparam file="$(find scout_unitree_control)/config/scout_tracking.yaml" command="load" />
    </node>

</launch>